src/dune_rules/merlin/merlin.ml

```diff
@@ -257,25 +286,33 @@ module Processed = struct
       | None -> Copy_line_directive.DB.follow_while file ~f:find
     in
     let pp = Module_name.Per_item.get pp_config (Module.name module_) in
-    to_sexp ~opens ~pp config
+    to_sexp ~opens ~pp (file_config config ~extension:(Path.Build.extension file))
   ;;
 
   let print_file path =
     match load_file path with
     | Error msg -> Printf.eprintf "%s\n" msg
     | Ok { per_module_config; pp_config; config } ->
-      let pp_one { module_; opens } =
+      let pp_one name sexp =
         let open Pp.O in
+        Pp.vbox (Pp.text name) ++ Pp.newline ++ Pp.vbox (Sexp.pp sexp)
+      in
+      let pp_module { module_; opens } =
         let name = Module.name module_ in
         let pp = Module_name.Per_item.get pp_config name in
-        let sexp = to_sexp ~opens ~pp config in
-        Pp.vbox (Pp.text (Module_name.to_string name))
-        ++ Pp.newline
-        ++ Pp.vbox (Sexp.pp sexp)
+        let sexps =
+          List.map (Module.sources module_) ~f:(fun path ->
```

Comment:
So this is working incorrectly now as it gets `.ml` suffixed versions, e.g for `App.mlx` it will get `App.mlx.ml` and `App.mlx.mli`.

1. I'm not sure how to fix that... seems like after `Module.ml_source` we don't track the original modules so no info about the dialect or original file name is present here.
2. As I understand this command is purely for tests/debug, maybe we should make it iterate over source files and "modules"?

Reply:
Just to clarify, in 2. above I'm suggesting to remove `print_file` completely and make `dune ocaml merlin dump-config` work by listing the directory and then querying a config for each source file found.

Reply:
I'm not familiar with this part of the code, @rgrinberg or @voodoos may know.

Reply:
I think this would simplified by moving the reader info the the part of the config file that is module-specific instead of the part that is common for all modules of a stanza.

See my comment here: https://github.com/ocaml/dune/pull/8567/files#r1314880623